### PR TITLE
Add CVE Details and Single-Key Filtering for JSON Output in safety scan

### PR DESCRIPTION
**Pull request by dylanpulver**
This PR introduces the following enhancements to the safety scan command:

**1. CVE Details in Detailed JSON Output**
* Added a new section, cve_details, to the JSON output when --detailed-output  is enabled and output format is specified to json.
* The cve_details section provides comprehensive information about CVEs, including affected packages, severity, CVE IDs, and advisory links.

**2. Single-Key Filtering for JSON Output**
* Implemented the ability to filter the JSON output by a single top-level key using the --filter option.
* Users can now focus on specific sections of the report, such as meta or cve_details, simplifying the output for targeted use cases.

**3. Improved Report Processing**
* Encapsulated the filtering logic into a reusable helper function, filter_json_keys, improving maintainability and readability.

**Testing**
* Manually tested JSON output with and without the --filter option to ensure correctness.
* Verified that:
    * JSON output includes cve_details only when --detailed-output is enabled.
    * The --filter option correctly outputs only the specified top-level key.

**Usage Examples**
* `python -m safety scan --detailed-output --output json`: Includes the new cve_details section in the JSON output.
* `python -m safety scan --detailed-output --output json --filter cve_details`: Outputs only the cve_details section.
* `python -m safety scan --detailed-output --output json --filter meta`: Outputs only the meta section.

These changes enhance the usability of the safety scan command by providing more actionable data and allowing users to filter output for specific use cases.

### CODE REVIEW

**Comment by jakub-safetycli** [comment]
Could you add a unit test for the generated JSON output.

**Comment by dylanpulver** [comment]
> Could you add a unit test for the generated JSON output.

Hey  @jakub-safetycli, I added unit tests that pass locally- the issue I am facing is that the cve_details new section is only added in the latest scan code which I run locally as a module

In the pre-existing test we invoke it via:
```
result = self.runner.invoke(cli, ["--stage", "cicd", "scan", "--target", self.dirname, "--output", "screen"])
```
For my new tests running this way does not contain the latest changes since it is not deployed. 

I switched for my tests to use subprocess via:
```
process = subprocess.run(
            ["python", "-m", "safety", "scan", "--target", ".", "--detailed-output", "--output", "json"],
            capture_output=True,
            text=True
        )
```
but in the CI output it is not able to parse the stdout.

Since the feature here is mostly straightforward and was easy to manually test I think it would be quicker to not get bogged down in this and add testing afterwards if it is strictly need. Please let me know what you think!

File: safety/scan/command.py [hunk 1]
```diff
@@ -250,7 +356,11 @@ def scan(ctx: typer.Context,
                             typer.Option("--apply-fixes",
                                 help=SCAN_APPLY_FIXES,
                                 show_default=False)
-                        ] = False
+                        ] = False,
+        filter_keys: Annotated[
```
Comment (jakub-safetycli) [inline safety/scan/command.py:360]: nit: Can you format this lie and line #578, the line wraps appear to be missing.

File: safety/scan/command.py [hunk 2]
```diff
@@ -175,6 +186,101 @@ def process_report(
...
+    """
+    Sorts CVE details by severity in descending order.
+
+    Args:
+        cve_data (List[Dict[str, Any]]): A list of CVE details dictionaries, each containing a 'severity' key.
+
+    Returns:
+        List[Dict[str, Any]]: The sorted list of CVE details, prioritized by severity (e.g., CRITICAL > HIGH > MEDIUM).
+    """
+    severity_order = {
```
Comment (jakub-safetycli) [inline safety/scan/command.py:230]: nit: We can probably relay on VulnerabilitySeverityLabels and do something like:

severity_order = {key.name: id for (id, key) in enumerate(VulnerabilitySeverityLabels)}
sorted(cve_data, key=lambda x: severity_order.get(x["severity"].upper(), 0), reverse=True)

File: safety/scan/command.py [hunk 3]
```diff
@@ -175,6 +186,101 @@ def process_report(
...
+    """
+    Filters and returns valid CVE details from a list of vulnerabilities.
+
+    Args:
+        vulnerabilities (List[Any]): A list of vulnerabilities, which may include invalid data types.
+
+    Returns:
+        List[Dict[str, Any]]: A list of filtered CVE details that are either strings or dictionaries.
+    """
+    return [
```
Comment (jakub-safetycli) [inline safety/scan/command.py:215]: Question: When the data might be invalid? Do you have any examples of those?

File: safety/scan/command.py [hunk 4]
```diff
@@ -162,6 +167,12 @@ def process_report(
 
         if output is ScanOutput.JSON or ScanOutput.is_format(output, ScanOutput.SPDX):
             if output is ScanOutput.JSON:
+                if detailed_output and output is ScanOutput.JSON:
```
Comment (jakub-safetycli) [inline safety/scan/command.py:170]: nit: `output is ScanOutput.JSON` should be always true.

Comment (dylanpulver) [reply safety/scan/command.py]: ah removed it

File: safety/scan/command.py (hunk 2 above)
Comment (dylanpulver) [reply safety/scan/command.py]: Ah switched now

File: safety/scan/command.py (hunk 1 above)
Comment (dylanpulver) [reply safety/scan/command.py:354]: I tried to fix this formatting, let me know if it got what we wanted

File: safety/scan/command.py (hunk 3 above)
Comment (dylanpulver) [reply safety/scan/command.py:215]: We don't have example that I know of, but this was just to make the code hardened

**Pull request merged**